src/gluonts/transform.py

```diff
@@ -345,14 +347,25 @@ class FlatMapTransformation(Transformation):
     def __call__(
         self, data_it: Iterator[DataEntry], is_train: bool
     ) -> Iterator:
+        num_idle_transforms = 0
         for data_entry in data_it:
+            num_idle_transforms += 1
             try:
                 for result in self.flatmap_transform(
                     data_entry.copy(), is_train
                 ):
+                    num_idle_transforms = 0
                     yield result
             except Exception as e:
                 raise e
+            if num_idle_transforms > GLUONTS_MAX_IDLE_TRANSFORMS:
```

Comment:
I think what I don't like, is that there is no way to turn this check off.

Reply:
Why would you want to switch it off? If you really want to you can set it to a very large number.